### PR TITLE
refactor(timm_mobilenetv3): migrate to Task SDK

### PR DESCRIPTION
**Pull request by yifeif-nv**
## Background

Migrate timm MobileNetV3 to the semantic Task SDK introduced by #1226.
Every changed file is under `families/timm_mobilenetv3/`; the shared C ABI,
C++ wrapper, CLI, benchmark and CI discovery already support this Task.

## Exit Criteria

- The family implements and advertises `image_to_class_scores`, returning every
  original logit in checkpoint class order through the C and C++ SDKs.
- Preserve the original graph, preprocessing and official timm top-1 oracle.
- Family CPU and single-device E2E checks pass. Required Pre-merge checks must
  pass before merge; local results are not a remote Pre-merge claim.

## Implementation

- Implement `IModel` and `IImageToClassScores`; the family returns its own
  `bind` record with no optional runtime Config fields.
- Update support, builder and the owned manifest to `image_to_class_scores`.
  Return complete owned logits, not a top-k subset or softmax probabilities.
- Carry class count and checkpoint-provided labels/vocabulary identity in the
  family's runtime metadata. Missing identity stays empty; no hash, invented
  name or local checkpoint path is used as a cross-model identity.
- Add CPU tests for binding, input shape, preprocessing, complete result ownership,
  known/unknown class identity and rejection of unsupported Config.
- Add direct public C11 and C++17 consumers to the existing single-device E2E.
  Both read the result after releasing the model and must agree on every score
  for identical RGB pixels. The original JPEG CLI path and timm top-1 check
  remain unchanged; different JPEG decoders are not claimed bitwise equivalent.
- Existing family CTest discovery builds the consumers through a target
  dependency. No shared registry, workflow or new environment variable is added.

### Follow-up: family-owned benchmark timing

The migration moved native argmax outside the timed Task call, while the old
reference still timed argmax and summary generation. The owning family now
provides `tests/performance.yaml` and `tests/performance_reference.py` through
the existing reference-script mechanism. Model inference, full host float32
scores and synchronization are timed; classification and reporting are not.

The original entry ID, workload, FP16 policy, 3 warmups, 10 measurements, 5%
margin and top-class oracle are preserved. The existing reference policy still
excludes input preparation whereas the native Task includes family preprocessing;
this change does not claim to remove that pre-existing scope difference or
establish a performance improvement. No shared file changes.

### Change categories

- [x] Model or runtime behavior
- [x] Public API
- [ ] ABI
- [x] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [x] CI or developer tooling

Public C layouts and bundle format 1 do not change. The family's primary Task
and runtime metadata do change: rebuild old classification bundles. There is
no old Task alias, compatibility adapter or fallback execution path.

### Commands and Results

The following original migration results belong to
`8d317e3e86cef6dda577694ca15aa082610b900f`, using the repository's native
dependencies and existing E2E environment. Follow-up validation is listed
separately below; these results are not new-head CI results.

- `cmake --build build --target trtmc trtmc_backend_trt test_timm_mobilenetv3_task_contract test_timm_mobilenetv3_image_preprocess`:
  passed, including the public C/C++ consumer binaries.
- `ctest --test-dir build --output-on-failure -R '^timm_mobilenetv3_(task_contract|image_preprocess)$'`:
  passed, 2/2 CPU tests.
- `python -m pytest families/timm_mobilenetv3/tests -m 'not gpu and not trt and not e2e' -q`:
  17 passed; the explicitly unselected real E2E case was skipped.
- `python -m pytest families/timm_mobilenetv3/tests/test_e2e.py --e2e-testcase mobilenetv3-large-100-ra-in1k -q`:
  1 real single-device case passed. CLI/C/C++ retain the original top-1 oracle;
  the two SDK consumers agree on all 1000 logits and class metadata.
- `python3 -m tools.community_ci source-quality --base ed98d6affe9e61be98ead407c1fd77e4155d6b75`:
  passed legal headers, inventory, complexity, formatting and 212 contract tests.
- `git diff --check`: passed. Existing benchmark resolution selects the new
  Task and retains its `classify` operation without a central suite edit.

Additional checks for the timing follow-up:

- `python3 -m pytest families/timm_mobilenetv3/tests/test_performance_reference.py -q`:
  15 passed, including timer ordering, receipt identity and invalid-output checks.
- The complete family CPU selection above now passes 32 tests, with only the
  explicitly unselected real E2E case skipped.
- The family reference executed the original single-GPU 3-warmup/10-measurement
  workload on the pinned checkpoint below: completed, 10 timing samples,
  1000 finite scores and top-class 656. This is reference execution evidence,
  not a new paired speedup or full-migration E2E claim.

### Hardware, Environment, and Revisions

- Base: `ed98d6affe9e61be98ead407c1fd77e4155d6b75` (#1226).
- Original migration E2E/native validation: `8d317e3e86cef6dda577694ca15aa082610b900f`.
- Timing follow-up CPU/reference validation: `bd007e2fe3fb859c4244b5efc0fc1f3c6d89a585`.
  Production runtime, builder and original E2E sources are unchanged by this follow-up.
- Linux/aarch64, one NVIDIA GB300 GPU, CUDA toolkit 13.3.33, TensorRT 11.1.0.106,
  PyTorch 2.12.0+cu130 and timm 1.0.28; FP16 as in the original manifest.
- Checkpoint: `timm/mobilenetv3_large_100.ra_in1k` revision
  `96f46a1c52932f27492dff66c72378eb99b443a7`; original checked-in test image.

### Not Run / Remaining Gaps

- No multi-device execution, full Nightly or all-checkpoint qualification.
- No claim of a performance improvement or cross-decoder bitwise parity.
- Required Pre-merge must pass on the follow-up head before merge; earlier-head
  results do not qualify this update.

## Contributor Self-Review

- [x] I have completed a self-review of this change.

Reviewed the family-only diff, unchanged preprocessing/reference, complete
outputs and public evidence, with independent technical and publication review.

## Notes For Future Readers

#1226 is already merged. This PR is independent of the PatchTSMixer migration
and other families. `families/timm_mobilenetv3/README.md` describes the new Task,
bundle rebuild and direct SDK validation commands. Runtime and family DSOs
upgrade together; obsolete shared execution interfaces are not removed here.

### Risk level

- [ ] Low
- [x] Medium
- [ ] High

This changes one family's public Task identity and bundle metadata. The original
model/CLI oracle and additional complete SDK-output checks constrain regression.

### CODE REVIEW

**Comment by yifeif-nv** [comment]
Review follow-up for `bd007e2fe3fb859c4244b5efc0fc1f3c6d89a585`:

The family-owned reference now places argmax and reporting after timing, while
timing inference and the complete host float32 result. The original workload,
3/10 sampling, 5% margin and top-class oracle are unchanged.

The remaining warning is narrower than a failed inference/parity result:

- Finite-value checks are outside timing on both paths: the native worker runs
  `json_values` in its observer, and the reference runs `_summary` after `_measure`.
- Float32 conversion is intentional. The native builder also casts engine output
  to float32; requiring the reference's raw tensor dtype to equal that transport
  dtype would incorrectly conflate execution precision with the public result.
- The reference builds the model with checkpoint `num_classes` and loads weights
  strictly, but does not repeat native's explicit output class-count guard.
  That additional defensive check is deferred as a non-blocking follow-up, not
  claimed implemented. The real reference run returned all 1000 finite scores
  and top class 656. No claim is made that metadata/ABI overhead is identical
  across implementations.

The existing preprocessing timing-scope difference remains documented in the
family README. No numerical criterion or required test has been waived; merge
readiness still requires the current head's protected Pre-merge gate.
